pulp_ansible/tests/functional/api/collection/v3/test_sync.py

```diff
@@ -203,6 +203,54 @@ def test_sync_with_missing_collection(self):
         msg = "absent.not_present does not exist"
         self.assertIn(msg, task_result["error"]["description"], task_result["error"]["description"])
 
+    def test_sync_with_proxy_auth(self):
+        """Test sync collections from pulp server."""
+        if not os.getenv("PULP_PROXY_TEST"):
+            self.skipTest("Proxy isn't set.")
+
+        second_body = gen_ansible_remote(
+            url=self.distribution.client_url,
+            requirements_file=self.requirements_file,
+            sync_dependencies=False,
+            proxy_url="http://127.0.0.1:8899",
+            proxy_username="foo",
+            proxy_password="bar",
+        )
+        second_remote = self.remote_collection_api.create(second_body)
+        self.addCleanup(self.remote_collection_api.delete, second_remote.pulp_href)
+
+        second_repo = self._create_repo_and_sync_with_remote(second_remote)
+
+        first_content = self.cv_api.list(
+            repository_version=f"{self.first_repo.pulp_href}versions/1/"
+        )
+        self.assertGreaterEqual(len(first_content.results), 1)
+        second_content = self.cv_api.list(repository_version=f"{second_repo.pulp_href}versions/1/")
+        self.assertGreaterEqual(len(second_content.results), 1)
+
+    def test_sync_with_proxy_auth_without_credentials(self):
+        """Test sync collections from pulp server."""
+        if not os.getenv("PULP_PROXY_TEST"):
+            self.skipTest("Proxy isn't set.")
+
+        second_body = gen_ansible_remote(
+            url=self.distribution.client_url,
+            requirements_file=self.requirements_file,
+            sync_dependencies=False,
+            proxy_url="http://127.0.0.1:8899",
+        )
+        second_remote = self.remote_collection_api.create(second_body)
+        self.addCleanup(self.remote_collection_api.delete, second_remote.pulp_href)
+
+        with self.assertRaises(PulpTaskError) as cm:
```

Comment:
What's the reason behind the name `cm`? If it's short for "context manager" i would rather call it `ctx` or `exc_ctx` because its the context managed by the context manager `self.assertRaises(PulpTaskError)`.

Reply:
I just copied the structure from another test